src/frontend/js/components/SaleTunnel/AddressSelector/index.tsx

```diff
@@ -0,0 +1,118 @@
+
+const messages = defineMessages({
+  label: {
+    defaultMessage: 'Billing address',
+    description: 'Address field label.',
+    id: 'components.SaleTunnel.AddressSelector.label',
+  },
+  edit: {
+    defaultMessage: 'Edit',
+    description: 'Address edit button.',
+    id: 'components.SaleTunnel.AddressSelector.edit',
+  },
+  create: {
+    defaultMessage: 'Create',
+    description: 'Address create button.',
+    id: 'components.SaleTunnel.AddressSelector.create',
+  },
+});
+
+export const AddressSelector = () => {
+  const intl = useIntl();
+  const addresses = useAddresses();
+  const { billingAddress, setBillingAddress } = useSaleTunnelContext();
+
+  const options = useMemo(
+    () =>
+      addresses.items.map((address) => ({
+        value: address.id,
+        label: getAddressLabel(address),
+      })),
+    [addresses.items],
+  );
+
+  // Set main address as default billing address
+  useEffect(() => {
+    if (!billingAddress) {
+      setBillingAddress(addresses.items.find((address) => address.is_main));
+    }
+  }, [addresses.items]);
+
+  const createFormModal = useModal();
+  const editFormModal = useModal();
+  const isMobile = useMatchMediaLg();
+
+  return (
+    <div className="mt-s sale-tunnel__information__billing-address">
+      <Select
+        label={intl.formatMessage(messages.label)}
+        options={options}
+        fullWidth
+        value={billingAddress?.id}
+        onChange={(e) => {
+          setBillingAddress(addresses.items.find((address) => address.id === e.target.value));
+        }}
+      />
+      {billingAddress ? (
+        <Button
+          size="small"
+          icon={<span className="material-icons">edit</span>}
+          color="tertiary"
+          onClick={editFormModal.open}
+          fullWidth={isMobile}
+        >
+          <FormattedMessage {...messages.edit} />
+        </Button>
+      ) : (
+        <Button
+          size="small"
+          icon={<span className="material-icons">add</span>}
+          color="primary"
+          onClick={createFormModal.open}
+          fullWidth={isMobile}
+        >
+          <FormattedMessage {...messages.create} />
+        </Button>
+      )}
+      <CreateAddressFormModal
+        {...createFormModal}
+        onSettled={(newAddress) => {
+          setBillingAddress(newAddress);
+          createFormModal.close();
+        }}
+      />
+      <EditAddressFormModal
+        {...editFormModal}
+        address={billingAddress!}
+        onSettled={(updatedAddress) => {
+          editFormModal.close();
+          setBillingAddress(updatedAddress);
+        }}
+      />
+    </div>
+  );
+};
+
+export function getAddressLabel(address: Address) {
+  const part = [
+    address.first_name,
+    address.last_name,
+    address.address,
+    address.postcode,
+    address.city,
+    address.country,
+  ].join(' ');
+  if (address.title) {
+    return `${address.title} - ${part}`;
+  }
+  return part;
+}
```

Comment:
What alternative can you think of? cc @jbpenrath

Reply:
It would be nice to have multiple lignes:
Home
12 rue de la cornich, 
paris 21 FRANCE
Work
13 quai des anglais, 
paris 31FRANCE

Reply:
Yup, it could be cool on two lines, but I think we will have to use a custom component to use a smaller font-size.

>**Work**
>13 quai des anglais, 75000 Paris - FR

>**Home**
>11 rue stine, 75000 Paris - FR